lib/segment/src/types.rs

```diff
@@ -705,6 +706,30 @@ pub struct StrictModeConfig {
 
 impl Eq for StrictModeConfig {}
 
+impl Hash for StrictModeConfig {
+    fn hash<H: std::hash::Hasher>(&self, state: &mut H) {
+        let Self {
+            enabled,
+            max_query_limit,
+            max_timeout,
+            unindexed_filtering_retrieve,
+            unindexed_filtering_update,
+            search_max_hnsw_ef,
+            search_allow_exact,
+            search_max_oversampling,
+        } = self;
+
+        enabled.hash(state);
+        max_query_limit.hash(state);
+        max_timeout.hash(state);
+        unindexed_filtering_retrieve.hash(state);
+        unindexed_filtering_update.hash(state);
+        search_max_hnsw_ef.hash(state);
+        search_allow_exact.hash(state);
+        search_max_oversampling.map(|i| i.to_le_bytes()).hash(state);
+    }
+}
```

Comment:
Can't we derive hash, or does `search_max_oversampling` make life difficult?

Otherwise we might want to not hash it to prevent float rounding errors from breaking hashes.

Reply:
Yes `search_max_oversampling` is the problem here.
We need `StrictModeConfig` to implement `Hash`, since it's [a field in this struct](https://github.com/qdrant/qdrant/blob/strict_mode_test/lib/storage/src/content_manager/collection_meta_ops.rs#L172), which derives `Hash`.

Reply:
Check.

Let's skip hashing `search_max_oversampling` because we cannot reliably do that:

```suggestion
impl Hash for StrictModeConfig {
    fn hash<H: std::hash::Hasher>(&self, state: &mut H) {
        let Self {
            enabled,
            max_query_limit,
            max_timeout,
            unindexed_filtering_retrieve,
            unindexed_filtering_update,
            search_max_hnsw_ef,
            search_allow_exact,
            // We skip hashing this field because we cannot reliably hash a float
            search_max_oversampling: _,
        } = self;

        (
            enabled,
            max_query_limit,
            max_timeout,
            unindexed_filtering_retrieve,
            unindexed_filtering_update,
            search_max_hnsw_ef,
            search_allow_exact,
        )
            .hash(state);
    }
}
```